index-handlers/src/main/java/no/sikt/nva/nvi/index/Aggregations.java

```diff
@@ -0,0 +1,66 @@
+package no.sikt.nva.nvi.index;
+
+import java.util.Map;
+import no.sikt.nva.nvi.index.model.ApprovalStatus;
+import org.opensearch.client.opensearch._types.aggregations.Aggregation;
+import org.opensearch.client.opensearch._types.query_dsl.BoolQuery.Builder;
+import org.opensearch.client.opensearch._types.query_dsl.Query;
+import org.opensearch.client.opensearch._types.query_dsl.QueryStringQuery;
+
+public final class Aggregations {
+
+    private static final String AFFILIATIONS = "affiliations";
+    private static final String APPROVAL_STATUS = "approvalStatus";
+    private static final String PENDING = "pending";
+    private static final String APPROVED = "approved";
+    private static final String REJECTED = "rejected";
+    private static final CharSequence JSON_PATH_DELIMITER = ".";
+    public static final Map<String, Aggregation> AGGREGATIONS = constructAggregations();
+
+    private Aggregations() {
+    }
+
+    private static String jsonPathOf(String... args) {
+        return String.join(JSON_PATH_DELIMITER, args);
+    }
+
+    private static Aggregation approvedAggregation() {
+        return new Aggregation.Builder()
+                   .filter(queryNotToMatchApprovalStatusPendingAndRejected())
+                   .build();
+    }
+
+    private static Query queryNotToMatchApprovalStatusPendingAndRejected() {
+        return new Builder()
+                   .mustNot(queryToMatchApprovalStatus(ApprovalStatus.PENDING),
+                            queryToMatchApprovalStatus(ApprovalStatus.REJECTED))
+                   .build()._toQuery();
+    }
+
+    private static Aggregation pendingAggregation() {
+        return new Aggregation.Builder()
+                   .filter(queryToMatchApprovalStatus(ApprovalStatus.PENDING))
+                   .build();
+    }
+
+    private static Aggregation rejectedAggregation() {
+        return new Aggregation.Builder()
+                   .filter(queryToMatchApprovalStatus(ApprovalStatus.REJECTED))
+                   .build();
+    }
+
+    private static Query queryToMatchApprovalStatus(ApprovalStatus approvalStatus) {
+        return new Query(new QueryStringQuery.Builder()
+                             .query(approvalStatus.toString())
+                             .defaultField(jsonPathOf(AFFILIATIONS, APPROVAL_STATUS))
+                             .build());
+    }
+
+    private static Map<String, Aggregation> constructAggregations() {
+        return Map.of(
+            jsonPathOf(APPROVAL_STATUS, PENDING), pendingAggregation(),
```

Comment:
Do we really have to explicitly mention all specific values to aggregate over?

Reply:
In this case yes, since we have do not have single status for each index document, but list of values for every affiliation. So to construct this aggregation I had actually to "filter out" statuses that are not relevant.
For example approvedAggregation() is aggregation for documents that do not contain statuses PENDING and REJECTED